client/my-sites/stats/post-trends/week.jsx

```diff
@@ -1,40 +1,70 @@
 /**
  * External dependencies
  */
-var i18n = require( 'i18n-calypso' ),
-	React = require( 'react' );
+import i18n from 'i18n-calypso';
+import React, { PropTypes } from 'react';
 
 /**
  * Internal dependencies
  */
-var Day = require( './day' );
+import Day from './day';
 
-module.exports = React.createClass( {
+export default React.createClass( {
 
 	displayName: 'PostTrendsWeek',
 
 	propTypes: {
-		startDate: React.PropTypes.object.isRequired,
-		month: React.PropTypes.object.isRequired,
-		data: React.PropTypes.object.isRequired
+		startDate: PropTypes.object.isRequired,
+		month: PropTypes.object.isRequired,
+		max: PropTypes.number,
+		streakData: PropTypes.object
 	},
 
-	getDayComponents: function() {
-		var i,
-			data = this.props.data.response.days || {},
-			max = this.props.data.response.max || 1,
-			days = [],
-			dayDate;
+	getDefaultProps() {
+		return {
+			streakData: {},
+			max: 0
+		};
+	},
+
+	getDayComponents() {
+		let days = [];
+		const { month, startDate, streakData, max } = this.props;
+
+		for ( let i = 0; i < 7; i++ ) {
+			const dayDate = i18n.moment( startDate ).add( i, 'day' );
+			const postCount = streakData[ dayDate.format( 'YYYY-MM-DD' ) ];
```

Comment:
I have a default prop value set to `0` for `postCount` on `<Day />`, but I suppose I could do that here too.